### PR TITLE
For Connection Updates, we need to use the correct connection type

### PR DESCRIPTION
**Pull request by kevinkirkup**

### CODE REVIEW

File: pureport/client/api_connections.go [hunk 1]
```diff
@@ -591,11 +591,30 @@ func (a *ConnectionsApiService) UpdateConnection(ctx context.Context, connection
 	// body params
 	if localVarOptionals != nil && localVarOptionals.Body.IsSet() {
 
-		localVarOptionalBody, localVarOptionalBodyok := localVarOptionals.Body.Value().(Connection)
-		if !localVarOptionalBodyok {
+		// ##################################################
+		// Pureport HACKAGE!!!
```
Comment (jmaness) [inline pureport/client/api_connections.go]: Will this be open source? If so, should this comment be omitted?

Comment (kevinkirkup) [reply pureport/client/api_connections.go]: It is open source. We can change the comments to something else, but we need to mark what was changed from swagger-code gen.

Comment (jmaness) [reply pureport/client/api_connections.go]: If we do need a delimiter (as opposed to git history), perhaps we can use something like `Pureport` or `Pureport (start)`/`Pureport (end)`. `HACKAGE!!!`, in my opinion, might send the wrong message to users/customers about the quality of the code.

Comment (kevinkirkup) [reply pureport/client/api_connections.go]: Sure...

Comment (kevinkirkup) [reply pureport/client/api_connections.go]: Updated

**Pull request merged**